src/services/expensesService.js

```diff
@@ -0,0 +1,72 @@
+const { Op } = require('sequelize');
+const { models } = require('../models/models.js');
+const { Expense } = models;
+
+const getAllExpenses = async (userId, categories, from, to) => {
+  const where = {};
+
+  if (userId) {
+    where.userId = userId;
+  }
+
+  if (categories) {
+    where.category = { [Op.in]: categories };
+  }
+
+  if (from) {
+    where.spentAt = { [Op.gte]: new Date(from) };
+  }
+
+  if (to) {
+    where.spentAt = {
+      ...where.spentAt,
+      [Op.lte]: new Date(to),
+    };
+  }
+
+  return Expense.findAll({ where });
+};
+
+const getExpensesById = async (id) => {
+  return Expense.findByPk(id);
+};
+
+const createExpense = async (
+  userId,
+  spentAt,
+  title,
+  amount,
+  category,
+  note,
+) => {
+  return Expense.create({
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  });
+};
+
+const deleteExpense = async (id) => {
+  await Expense.destroy({
+    where: {
+      id,
+    },
+  });
+};
+
+const updateExpense = async ({ id, title }) => {
```

Comment:
The `updateExpense` function currently only updates the `title` field. Consider allowing updates to other fields if needed, and ensure that the update logic aligns with the application's requirements.